strawberry/experimental/pydantic/utils.py

```diff
@@ -85,9 +85,11 @@ def defaults_into_factory(
     final_factory = default_factory
     if default is not UNSET:
         if default_factory is not UNSET:
-            raise BothDefaultAndDefaultFactoryDefinedError(
-                default=default, default_factory=default_factory  # type: ignore
-            )
+            # pydantic ModelField default and default_factory are None by default
+            if default is not None and default_factory is not None:
+                raise BothDefaultAndDefaultFactoryDefinedError(
+                    default=default, default_factory=default_factory  # type: ignore
+                )
```

Comment:
hacky?

if you looked at pydantic's ModelField, it sets default and default_factory to None by default.
rather than having a value such as UNSET to differentiate between None as the user's default, and having no default.